src/screens/LoginScreen.js

```diff
@@ -24,40 +23,31 @@ const {campo_libre, tagline, login_as_guest, login_with_facebook} = login;
 class LoginScreen extends Component {
 
     componentWillMount() {
-        this.props.checkAndSetToken();
-    }
-
-    componentWillReceiveProps(nextProps) {
-        const {token, navigation: {navigate}} = nextProps;
+        const {navigation: {navigate}} = this.props;
 
-        if (token) {
-            navigate(navKeys.SEARCH);
-        }
```

Comment:
The logic for navigation can be taken out since we're explicitly doing it with the actions now.